openfoodfacts-implementation/data-generation.ipynb

```diff
@@ -0,0 +1,299 @@
+    "    \"\"\"\n",
+    "    # We filter the Dask DataFrame by the ingredient name.\n",
+    "    # This way, we will only be left with the rows that contain the ingredient in their name.\n",
+    "    filtered = ddf[ddf['generic_name'].str.contains(ingredient, case=False, na=False)]\n",
+    "    \n",
+    "    # Select nutritional columns\n",
+    "    nutrition_columns = ['energy_100g', 'sugars_100g', 'fiber_100g', 'fat_100g', 'proteins_100g', 'carbohydrates_100g']\n",
+    "    \n",
+    "    # Calculate the average values\n",
+    "    average_nutrition = filtered[nutrition_columns].mean().compute()\n",
+    "    \n",
+    "    return average_nutrition\n",
+    "\n",
+    "def get_recipe_nutrition(ddf: dd.DataFrame, recipe: dict) -> pd.Series:\n",
+    "    \"\"\"Function to obtain the nutritional values of a recipe\n",
+    "\n",
+    "    Args:\n",
+    "        ddf (dd.DataFrame): Dask DataFrame with nutritional information of foods\n",
+    "        recipe (dict): Recipe with ingredients and portion\n",
+    "        \n",
+    "    Returns:\n",
+    "        pd.Series: Nutritional values of the recipe\n",
+    "    \"\"\"\n",
+    "    ingredients = recipe['ingredients']\n",
+    "    portion_size = recipe['portion']\n",
+    "    \n",
+    "    total_nutrition = pd.Series(0, index=['energy_100g', 'sugars_100g', 'fiber_100g', 'fat_100g', 'proteins_100g', 'carbohydrates_100g'])\n",
+    "    count = 0\n",
+    "    \n",
+    "    # For each ingredient in the dish, we calculate the average nutritional values\n",
+    "    for ingredient in ingredients:\n",
+    "        average_nutrition = get_average_nutrition(ddf, ingredient)\n",
+    "        # If the ingredient has no nutritional values, we do not count it\n",
+    "        if not average_nutrition.isnull().any():\n",
+    "            total_nutrition += average_nutrition\n",
+    "            count += 1\n",
+    "    \n",
+    "    # If there is more than one ingredient, we calculate the average nutritional values\n",
+    "    if count > 0:\n",
+    "        average_nutrition = total_nutrition / count\n",
+    "    else:\n",
+    "        average_nutrition = total_nutrition\n",
+    "    \n",
+    "    # Adjust nutritional values ​​according to serving\n",
+    "    multiplier = portion_size / 100\n",
+    "    recipe_nutrition = average_nutrition * multiplier\n",
+    "    \n",
+    "    return recipe_nutrition"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Calculate nutritional values ​​for each recipe in the JSON\n",
+    "for recipe_name, recipe_info in food_info.items():\n",
+    "    nutrition = get_recipe_nutrition(ddf, recipe_info)\n",
+    "    food_info[recipe_name]['nutrition'] = nutrition.to_dict()\n",
+    "\n",
+    "# Save the updated JSON\n",
+    "with open('./data/calculated_food_info.json', 'w') as f:\n",
+    "    json.dump(food_info, f, indent=4)"
+   ]
```

Comment:
Is it really necessary to save this intermediate step? If it's needed only for Google Collab, then remove it from the `.py` file when you make it.